modules/engine-prototype/src/main/scala/engine/Execution.scala

```diff
@@ -0,0 +1,17 @@
+package engine
+
+import cats.effect.Effect
+
+object Execution {
+
+  def configureTCS[F[_]](implicit F: Effect[F]): F[Either[Failure, OK]]  = ???
+
+  def configureInst[F[_]](name: String)(implicit F: Effect[F]): F[Either[Failure, OK]]  = ???
+
+  def observe[F[_]](implicit F: Effect[F]): F[Option[String]] = ???
+
+  type Failure = Unit
+
+  type OK = Unit
+
+}
```

Comment:
These functions are at the boundary with the underlying EPICS subsystems . Equivalent to what we call `Action`s in the current seqexec. They are supposed to be part of the *mini-DSL* to define Step executions. Obviously, they'll be shared among the different Steps so that's why they are defined in a separate object.

Reply:
They are scary at first, but not that bad after you get used to read them. Normal `IO` would be the primary instance of those type classes, but you are right, you could run this code with a fake `IO` type that captures all side effects for testing without doing actually any real IO.